### PR TITLE
Number data conversion tests failures exception fix for varchar accessor.

### PR DESCRIPTION
**Pull request by escobargabriel**
Make throw SQL exceptions for varchar accessors.

### CODE REVIEW

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/complex/AbstractArrowFlightJdbcUnionVectorAccessorTest.java [hunk 1]
```diff
@@ -225,9 +225,13 @@ public void testGetObjectClassUsesSpecificAccessor() {
...
 
   @Test
-  public void testGetObjectWithClassUsesSpecificAccessor() {
-    accessor.getObject(Object.class);
-    verify(innerAccessor).getObject(Object.class);
+  public void testGetObjectWithClassUsesSpecificAccessor() throws SQLException {
+    try {
+      accessor.getObject(Object.class);
+      verify(innerAccessor).getObject(Object.class);
+    } catch (Exception e) {
```
Comment (rafael-telles) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/complex/AbstractArrowFlightJdbcUnionVectorAccessorTest.java:232]: Unnecessary try/catch

**Review by rafael-telles — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessorTest.java [hunk 2]
```diff
@@ -188,43 +188,43 @@ public void testShouldFailToGetBoolean() {
...
   }
 
   @Test(expected = UnsupportedOperationException.class)
-  public void testShouldFailToGetDouble() {
+  public void testShouldFailToGetDouble() throws SQLException {
     when(accessor.getDouble()).thenCallRealMethod();
     accessor.getDouble();
   }
 
   @Test(expected = UnsupportedOperationException.class)
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessorTest.java:226]: Is it correct? Are we expecting UnsupportedOperationExcetion,class?

**Review by jcralmeida — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/complex/AbstractArrowFlightJdbcUnionVectorAccessorTest.java [hunk 3]
```diff
@@ -225,9 +200,10 @@ public void testGetObjectClassUsesSpecificAccessor() {
   }
 
   @Test
-  public void testGetObjectWithClassUsesSpecificAccessor() {
+  public void testGetObjectWithClassUsesSpecificAccessor() throws SQLException {
     accessor.getObject(Object.class);
     verify(innerAccessor).getObject(Object.class);
+
```
[the remark this answers is not in the document]
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/complex/AbstractArrowFlightJdbcUnionVectorAccessorTest.java:206]: Done.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessorTest.java (hunk 2 above)
Comment (escobargabriel) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessorTest.java:226]: Done.

**Review by jcralmeida — APPROVED** [review]

**Pull request merged**